src/modules/SpaceCalculation.rb

```diff
@@ -578,183 +555,149 @@ def get_partition_info
+              free_size -= used
+              growonly = btrfs_snapshots?(tmpdir)
+            else
+              partition = SCR.Read(path(".run.df"))
+
+              Builtins.foreach(partition) do |p|
+                if p["name"] == tmpdir
+                  log.info "Partition: #{p}"
+                  free_size = p["free"].to_i * 1024
+                  used = p["used"].to_i * 1024
+                end
               end
             end
+
+            SCR.Execute(path(".target.bash"), "umount #{Shellwords.escape(tmpdir)}")
+          else
+            log.error "Mount failed, ignoring partition #{device}"
+            @failed_mounts << filesystem
+
+            next
+          end
+        else
+          # for formatted partitions estimate free system size
+          # compute fs overhead
+          used = EstimateFsOverhead(filesystem)
+          device = filesystem_name(filesystem)
+          log.info "#{device}: assuming fs overhead: #{used / 1024}KiB"
+
+          # get the journal size
+          if type?(:extX, filesystem)
+            js = ExtJournalSize(filesystem)
+            reserved = ReservedSpace(filesystem)
+            used += reserved if reserved > 0
+          elsif type?(:xfs, filesystem)
+            js = XfsJournalSize(filesystem)
+          elsif type?(:reiserfs, filesystem)
+            js = ReiserJournalSize(filesystem)
+          elsif type?(:jfs, filesystem)
+            js = JfsJournalSize(filesystem)
+          elsif type?(:btrfs, filesystem)
+            # Btrfs uses temporary trees instead of a fixed journal,
+            # there is no journal, it's a logging FS
+            # http://en.wikipedia.org/wiki/Btrfs#Log_tree
+            js = 0
+          else
+            log.warn "Unknown journal size for filesystem: #{filesystem.type}"
+          end
+
+          if js && js > 0
+            log.info "Partition #{device}: assuming journal size: #{js / 1024}KiB"
+            used += js
+          end
+
+          # decrease free size
+          free_size -= used
+
+          # check for underflow
+          if free_size < 0
+            log.info "Fixing free size: #{free_size} to 0"
+            free_size = 0
           end
-        end # foreach (`part)
-      end # foreach (`disk)
+        end
+
+        # convert into KiB for TargetInitDU
+        free_size_kib = free_size / 1024
+        used_kib = used / 1024
+        mount_name = filesystem_mountpoint(filesystem)
+        log.info "partition: mount: #{mount_name}, free: #{free_size_kib}KiB, used: #{used_kib}KiB"
+
+        mount_name = mount_name[1..-1] if remove_slash && mount_name != "/"
+
+        target_partitions << {
+          # FIXME: in the pre-storage-ng era, this used to be the short name of
+          # the type (like "ext3"). Now it's a number (libstorage-ng enum).
+          # Not easy to fix without relying in yast2-storage-ng (which contains
+          # the enum mappings).
+          "filesystem" => filesystem.type,
```

Comment:
@shundhammer, @aschnell The problem explained in the comment is probably another symptom of something that has been pointed a couple of times before (specially by @jreidinger). The `filesystem.type` provided by libstorage should probably be something more rich than an enum. If it would be an object of `Storage::FilesystemType` (for example), we could do things like `filesystem.type.short_name`. This code is supposed to avoid relying in yast2-storage.

If you take a look to the `type?` method below, you can see more examples of the kind of things it would make sense to query to such an `FilesystemType` object.